lib/std/private/strimpl.nim

```diff
@@ -0,0 +1,45 @@
+func toLowerAscii*(c: char): char {.inline.} =
+  if c in {'A'..'Z'}:
+    result = chr(ord(c) + (ord('a') - ord('A')))
+  else:
+    result = c
+
+template cmpIgnoreStyleImpl*(a, b: typed, caseSensitive: static bool = false) =
+  # a, b are string or cstring
+  let aLen = a.len
+  let bLen = b.len
+  when caseSensitive:
+    if a[0] != b[0]: return 1
```

Comment:
```suggestion
    if a[0] != b[0]: return ord(a[0]) - ord(b[0])
```
see https://github.com/nim-lang/Nim/pull/16392#discussion_r546056332

Reply:
It is not correct. It needs to consider the boundary of string. cstring differs in JS and C backend(null terminator which the reason why I say this function is hard to be used in C backend).